Sources/VitalHealthKit/HealthKit/VitalHealthKitClient.swift

```diff
@@ -224,6 +224,25 @@ extension VitalHealthKitClient {
 
     self.backgroundDeliveryTask = Task(priority: .high) {
       for await payload in stream {
+        // If the task is cancelled, we would break the endless iteration and end the task.
+        // Any buffered payload would not be processed, and is expected to be redelivered by
+        // HealthKit.
+        //
+        // https://developer.apple.com/documentation/healthkit/hkhealthstore/1614175-enablebackgrounddelivery
+        try Task.checkCancellation()
```

Comment:
Wrong link?

Reply:
No exception is thrown. This throws the standard `CancellationError()` to end the loop early, and the task's result would become `.failure(CancellationError())`.

We don't care about the `backgroundDeliveryTask`'s result, since we use the task handle only as a cancellable. So that's pretty much it.